### PR TITLE
NEM

### PR DESCRIPTION
**Pull request by tsusanka**
To be reviewed, now complete

### CODE REVIEW

File: embed/extmod/modtrezorcrypto/modtrezorcrypto-bip32.h [hunk 1]
```diff
@@ -311,6 +313,57 @@ STATIC mp_obj_t mod_trezorcrypto_HDNode_address(mp_obj_t self, mp_obj_t version)
...
 
+/// def nem_address(self, network: int) -> str:
+///     '''
+///     Compute a NEM address string from the HD node.
+///     '''
+STATIC mp_obj_t mod_trezorcrypto_HDNode_nem_address(mp_obj_t self, mp_obj_t network) {
+    mp_obj_HDNode_t *o = MP_OBJ_TO_PTR(self);
+
+    uint8_t n = mp_obj_get_int_truncated(network);
+    char address[NEM_ADDRESS_SIZE];
```
[the remark this answers is not in the document]
Comment (tsusanka) [reply embed/extmod/modtrezorcrypto/modtrezorcrypto-bip32.h]: Oh, right. Does this mean the `\0` string termination byte?

File: embed/extmod/modtrezorcrypto/modtrezorcrypto-bip32.h [hunk 2]
```diff
@@ -311,6 +313,57 @@ STATIC mp_obj_t mod_trezorcrypto_HDNode_address(mp_obj_t self, mp_obj_t version)
...
+    hdnode_get_nem_address(&o->hdnode, n, address);
```
[the remark this answers is not in the document]
Comment (tsusanka) [reply embed/extmod/modtrezorcrypto/modtrezorcrypto-bip32.h]: I'll add it here then as well, right? https://github.com/trezor/trezor-core/blob/6e7a15c088d621cd05ec800fa35804122ce6db8e/embed/extmod/modtrezorcrypto/modtrezorcrypto-bip32.h#L309

[the remark this answers is not in the document]
Comment (tsusanka) [reply embed/extmod/modtrezorcrypto/modtrezorcrypto-bip32.h]: :+1:

**Comment by tsusanka** [comment]
Just for reference: closes #142

File: src/apps/nem/mosaic/layout.py [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
-from trezor.messages import NEMSupplyChangeType
-from trezor.messages import NEMMosaicDefinition
-from trezor.messages import NEMMosaicCreation
-from trezor.messages import NEMMosaicSupplyChange
-from trezor.messages import NEMMosaicLevy
+from micropython import const
+from trezor import ui
+from trezor.messages import (NEMMosaicCreation, NEMMosaicDefinition,
+                             NEMMosaicLevy, NEMMosaicSupplyChange,
+                             NEMSupplyChangeType, NEMTransactionCommon)
```
Comment (tsusanka) [inline src/apps/nem/mosaic/layout.py]: @jpochyla Not that I'm very keen on this, but leaving one import on one line makes cleaner diffs imho.

Comment (jpochyla) [reply src/apps/nem/mosaic/layout.py]: I'm not either! Lets revisit style later, across the whole codebase.

**Comment by prusnak** [comment]
:tada:

**Pull request merged**